internal/langserver/handlers/completion_hooks.go

```diff
@@ -1,15 +1,25 @@
 package handlers
 
 import (
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/search"
 	"github.com/hashicorp/hcl-lang/decoder"
+	"github.com/hashicorp/terraform-ls/internal/algolia"
 	"github.com/hashicorp/terraform-ls/internal/hooks"
 )
 
-func (s *service) AppendCompletionHooks(ctx decoder.DecoderContext) {
+func (s *service) AppendCompletionHooks(decoderContext decoder.DecoderContext) {
 	h := hooks.Hooks{
-		ModStore: s.modStore,
+		ModStore:       s.modStore,
+		RegistryClient: s.registryClient,
+		Logger:         s.logger,
 	}
 
-	ctx.CompletionHooks["CompleteLocalModuleSources"] = h.LocalModuleSources
+	credentials, ok := algolia.CredentialsFromContext(s.srvCtx)
+	if ok {
+		h.AlgoliaClient = search.NewClient(credentials.AppID, credentials.APIKey)
```

Comment:
Nitpick: I suppose we could also avoid registering the completion hook if the credentials aren't available?

Reply:
Interesting idea. I think the code looks a bit cleaner if we always register the hook and have it exiting early. So it's up to the hook to decide if all requirements are met for a run.